[API] Specify optional mail notification

NobodysNightmare opened the pull request:
Specifies:
- [#4128](https://community.openproject.org/work_packages/4128)
- [#17009](https://community.openproject.org/work_packages/17009) (dup)

NobodysNightmare committed:
specify option to limit notifications

[ci skip]

TeatroIO commented:
I've prepared a stage. <a href='http://feature-specify-optional-mail-notification.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com' target='_blank'>Click to open.</a>

ulferts commented:
I am fine with that one (have only been looking at last commit).

NobodysNightmare committed:
more concise parameter names

[ci skip]

NobodysNightmare commented:
I rebased this PR onto the dev as there was really no good reason to branch from somewhere else.

Also incorporated renaming proposed in https://github.com/opf/openproject/commit/06f911f2a7376955b4e92791c4d70ac400a2e046 (before rebasing).
